Library/Homebrew/utils/curl.rb

```diff
@@ -6,8 +6,8 @@ def curl_executable
     ENV["HOMEBREW_CURL"],
     which("curl"),
     "/usr/bin/curl",
-  ].map { |c| Pathname(c) }.find(&:executable?)
-  raise "curl is not executable" unless @curl
+  ].reject(&:nil?).map { |c| Pathname(c) }.find(&:executable?)
```

Comment:
`.reject(&:nil?)` should be `.compact`

Reply:
My ruby-fu is quite weak. I've changed it but haven't tested.